src/utils/__tests__/async.test.ts

```diff
@@ -4,6 +4,41 @@ import { logger } from '../../logger';
 
 jest.mock('../../logger');
 
+import { retrySpawnProcess } from '../async';
+import { spawnProcess } from '../system';
+
+jest.mock('../system', () => {
+  const original = jest.requireActual('../system');
+  return {
+    ...original,
+    spawnProcess: jest.fn(original.spawnProcess),
+  };
```

Comment:
I don't think we need this dance here. Fully mocking the `system` module should be enough.

Reply:
We need it for this test: https://github.com/getsentry/craft/blob/iker/feat%2Fmaven-target/src/utils/__tests__/async.test.ts#L23-L28.